public/app/features/alerting/EditNGAlert.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { PureComponent } from 'react';
+import { hot } from 'react-hot-loader';
+import { MapDispatchToProps, MapStateToProps } from 'react-redux';
+import { Spinner } from '@grafana/ui';
+import { connectWithCleanUp } from '../../core/components/connectWithCleanUp';
+import NextGenAlertingPage from './NextGenAlertingPage';
+import { getAlertDefinition, updateAlertDefinition } from './state/actions';
+import { getRouteParamsId } from '../../core/selectors/location';
+import { AlertDefinition, StoreState } from '../../types';
+
+interface OwnProps {}
+
+interface ConnectedProps {
+  alertDefinition: AlertDefinition;
+  pageId: string;
+}
+
+interface DispatchProps {
+  getAlertDefinition: typeof getAlertDefinition;
+  updateAlertDefinition: typeof updateAlertDefinition;
+}
+
+type Props = ConnectedProps & DispatchProps & OwnProps;
+
+export class EditNGAlert extends PureComponent<Props> {
+  componentDidMount() {
+    const { pageId, getAlertDefinition } = this.props;
+    getAlertDefinition(pageId);
+  }
+
+  render() {
+    const { alertDefinition, updateAlertDefinition } = this.props;
+
+    if (alertDefinition.id < 1) {
+      return <Spinner />;
+    }
```

Comment:
@peterholmberg why not using an additional loading state property rather than relying on the id? Feels like a pottential source of problems/confussion when the contract changes.